frontend/public/components/nav/perspective-nav.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import * as _ from 'lodash-es';
+import { createLink } from './items';
+import { NavSection } from './section';
+import AdminNav from './admin-nav';
+import { getActivePerspective } from '../../reducers/ui';
+import { RootState } from '../../redux';
+import * as plugins from '../../plugins';
+
+type StateProps = {
+  perspective: string;
+};
+
+const PerspectiveNav: React.FC<StateProps> = ({ perspective }) => {
+  // Until admin perspective is contributed through extensions, simply render static `AdminNav`
+  if (perspective === 'admin') {
+    return <AdminNav />;
+  }
+
+  // track sections so that we do not create duplicates
+  const renderedSections: string[] = [];
```

Comment:
This means a non-`admin` perspective always starts with empty nav menu, to be populated (owned) by Console plugin(s).

Do we want to enforce "perspective X owned by a single plugin" strategy? (In accordance with the general rule that plugins are idependent of each other.)

Reply:
Correct. If we can build the entire nav through extensions, I suggest that we do so instead of mixing static + extensions.

I see no issue with another plugin contributing to the perspective nav of another. For example a plugin may contribute to admin console and dev console, but if dev console isn't available, their extensions simply become inactive.

Reply:
> If we can build the entire nav through extensions, I suggest that we do so instead of mixing static + extensions.

:+1: 

> For example a plugin may contribute to admin console and dev console, but if dev console isn't available, their extensions simply become inactive.

Makes sense, agreed.